charmcraft/providers/_buildd.py

```diff
@@ -109,5 +107,28 @@ def setup(
         :raises BaseConfigurationError: on other unexpected error.
         """
         super().setup(executor=executor, retry_wait=retry_wait, timeout=timeout)
+        self._setup_charmcraft(executor=executor)
+
+    def warmup(
```

Comment:
Is this method actually used at runtime or is it provided only for testing convenience? I don't see it being called by the application.